### ISSUE TITLE
Unable to save document on Windows

### ISSUE THREAD

**Issue by gopaljigaur** [tinacms/tinacms#2810]
## Description

I integrated Tina into my existing Next.js project using this tutorial: [Adding Tina to an existing NextJS site](https://tina.io/docs/introduction/tina-init/). I get the message `There was a problem saving your document` on the UI when I try to edit the provided `HelloWorld.md` from Tina.

### Steps to reproduce

- Initialize a Next.js project (`npx create-next-app@latest`)
- Add TinaCMS to the project (`npx @tinacms/cli@latest init`)
- Start development server (`yarn dev`)
- Navigate to https://localhost:3000/admin and authenticate Tina
- Navigate to the demo page: http://localhost:3000/demo/blog/HelloWorld
- Make some edits and click 'Save'

### Expected result

The document should be saved successfully without any errors.

### Actual result

- You'll be presented with _There was a problem saving your document_ when trying to save the document
- This is the error in logs:
```
[
  TypeError: Cannot read properties of undefined (reading 'name')
      at Database.put (D:\project\node_modules\@tinacms\graphql\dist\index.js:23643:34)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Resolver.updateResolveDocument (D:\project\node_modules\@tinacms\graphql\dist\index.js:22633:7)
      at async fieldResolver (D:\project\node_modules\@tinacms\graphql\dist\index.js:23228:37)
      at async Object.resolve (D:\project\node_modules\@tinacms\graphql\dist\index.js:23112:17)
      at async D:\project\node_modules\@tinacms\cli\dist\index.js:95:24 {
    locations: [ [Object] ],
    path: [ 'updatePostsDocument' ]
  }
]
```

Looking into the source file of the error (`D:\project\node_modules\@tinacms\graphql\dist\index.js`), this error arises in the following code:

```
await this.store.put(filepath, payload, {
          keepTemplateKey,
          collection: collection.name,
          indexDefinitions: collectionIndexDefinitions
        });
```
`collection` is undefined during the database put request. So, `collection.name` cannot be referenced. A simple ```if(collection){...}``` check before execution might fix this.

#### Tinacms Package Versions

```text
﻿+-- @tinacms/cli@0.60.11
+-- tinacms@0.67.0
```

#### Tools

| Tool    | Version |
| ------- | ------- |
| browser | Microsoft Edge Version 100.0.1185.39 (Official build) (64-bit) |
| os      | Windows 11 Version 21H2 OS Build 22000.613 |
| node    | v16.13.2 |
| npm     | 7.24.1 |

**Comment by kldavis4** [comment]
@gopaljigaur are you using `yarn dev` or `next dev` to start the dev server? If you are running `next dev` without `tinacms`, the graphql api will not be running, but you shouldn't even be able to load the HelloWorld page without it.

(`yarn dev` executes: `tinacms server:start -c "next dev"`)

**Comment by gopaljigaur** [comment]
> @gopaljigaur are you using `yarn dev` or `next dev` to start the dev server? If you are running `next dev` without `tinacms`, the graphql api will not be running, but you shouldn't even be able to load the HelloWorld page without it.
> 
> (`yarn dev` executes: `tinacms server:start -c "next dev"`)

Sorry for that my mistake. I am actually using `yarn dev` to start the server. I have edited the issue to include that

**Comment by kldavis4** [comment]
@gopaljigaur thanks for the clarification. I was able to reproduce this in a windows environment. I'll do some investigation and update the ticket once I determine the issue.

**Comment by gopaljigaur** [comment]
@kldavis4 I don't know if it is an acceptable fix, but in `D:\project\node_modules\@tinacms\graphql\dist\index.js`, changing this:
```
await this.store.put(filepath, payload, {
          keepTemplateKey,
          collection: collection.name,
          indexDefinitions: collectionIndexDefinitions
        });
```

to this:

```
if(collections){
    await this.store.put(filepath, payload, {
          keepTemplateKey,
          collection: collection.name,
          indexDefinitions: collectionIndexDefinitions
        });
}
```

fixed the issue and I was able to save the document.

**Comment by kldavis4** [comment]
unfortunately that doesn't really address the root cause and will prevent the data layer from functioning properly. 

As a workaround, if you edit the generated `.tina/schema.ts` and find all the `path` fields and replace the forward slashes in the path with backward slashes, that should workaround the issue.

For example: `path: "content/posts",` -> `path: "content\posts",`

**Comment by gopaljigaur** [comment]
This issue was resolved after #2815
